File: MATTHEW/leetcode/AIcoding/5.Qwen-3/cpp_solutions/2923-find-champion-i.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        vector<int> inDegree(n, 0);
        for (const auto& edge : edges) {
            inDegree[edge[1]]++;
        }

        for (int i = 0; i < n; ++i) {
            if (inDegree[i] == 0) {
                return i;
            }
        }

        return -1;
    }
};

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/rust_solutions/2923-find-champion-i.rs
impl Solution {

use std::cmp::Ordering;

impl Solution {
    pub fn find_champion(n: i32, edges: Vec<Vec<i32>>) -> i32 {
        let mut in_degree = vec![0; n as usize];
        for edge in edges {
            in_degree[edge[1] as usize] += 1;
        }

        let mut candidates = Vec::new();
        for i in 0..n as usize {
            if in_degree[i] == 0 {
                candidates.push(i as i32);
            }
        }

        match candidates.len() {
            1 => candidates[0],
            _ => -1,
        }
    }
}
}